src/components/IntegrationTest/IntegrationTestParams.tsx

```diff
@@ -0,0 +1,323 @@
+  DataListItemRow,
+  DataListItemCells,
+  DataListCell,
+  CardHeader,
+  Card,
+  CardExpandableContent,
+  CardBody,
+} from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/js/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/js/icons/plus-circle-icon';
+import { useField } from 'formik';
+import { InputField } from '../../shared';
+import { Param } from '../../types/coreBuildService';
+
+interface IntegrationTestParamsProps {
+  heading?: string;
+  fieldName: string;
+  initExpanded?: boolean;
+}
+
+const IntegrationTestParams: React.FC<IntegrationTestParamsProps> = ({
+  heading,
+  fieldName,
+  initExpanded = false,
+}) => {
+  const [, { value: parameters, touched, error }, { setValue, setTouched }] =
+    useField<Param[]>(fieldName);
+
+  const initExpandedState = React.useMemo(() => {
+    const state = [];
+    for (let i = 0; i < parameters?.length; i++) {
+      state.push(false);
+    }
+    return state;
+  }, [parameters]);
+
+  const [expanded, setExpanded] = React.useState<boolean[]>(initExpandedState);
+  const [paramExpanded, setParamExpanded] = React.useState<boolean>(initExpanded);
+
+  const removeParam = (i) => {
+    if (!touched) {
+      setTouched(true);
+    }
+    const currentParam = [...parameters];
+    currentParam.splice(i, 1);
+    const expandedState = [...expanded];
+    expandedState.splice(i, 1);
+    setValue(currentParam);
+    setExpanded(expandedState);
+  };
+
+  const addParam = () => {
+    if (!touched) {
+      setTouched(true);
+    }
+    if (!parameters) {
+      setValue([{ name: 'param1', value: '' }]);
+      setExpanded([true]);
+      return;
+    }
+    setValue([...parameters, { name: `param${parameters.length + 1}`, value: '' }]);
+    setExpanded([...expanded, true]);
+  };
+
+  const addValue = (i) => {
+    if (!touched) {
+      setTouched(true);
+    }
+    const currentParams = [...parameters];
+
+    if (currentParams[i].value !== null) {
+      currentParams[i].values = [currentParams[i].value, ''];
+      currentParams[i].value = null;
+      setValue(currentParams);
+      return;
+    }
+    if (Array.isArray(parameters[i].values) && parameters[i].values.length > 0) {
+      currentParams[i].values = [...parameters[i].values, ''];
+      setValue(currentParams);
+      return;
+    }
+  };
+
+  const removeValue = (i, j) => {
+    if (!touched) {
+      setTouched(true);
+    }
+    const currentParams = [...parameters];
+
+    if (Array.isArray(parameters[i].values) && parameters[i].values.length > 1) {
+      (currentParams[i].values as string[]).splice(j, 1);
+      setValue(currentParams);
+      return;
+    }
+    if (Array.isArray(parameters[i].values) && parameters[i].values.length === 1) {
+      currentParams[i].value = currentParams[i].values[0];
+      currentParams[i].values = null;
+      setValue(currentParams);
+      return;
+    }
+  };
+
+  const toggleExpandedState = (i) => {
```

Comment:
Isn't there a uncontrolled version of expand collapse section which handles the state automatically?

Reply:
I am not sure, but I also need to expand new params when created.